src/commands/tasks/util.ts

```diff
@@ -25,3 +30,154 @@ export async function zipFileFromDirectory(path: string): Promise<string> {
     archive.finalize();
   });
 }
+
+export async function createFunctionUser(sdk: OAuth1Client, data: { taskName: string; targetEmail?: string; targetPermissions: string[]; }) {
+  const { taskName, targetEmail, targetPermissions } = data;
+
+  const email = targetEmail || `exh.tasks+${taskName}@extrahorizon.com`;
+  validateEmail(email);
+
+  const password = `A-${uuidv4()}-z`;
+
+  const roleName = `exh.tasks.${taskName}`;
+  const role = await createRoleWithPermissions(sdk, taskName, roleName, targetPermissions);
+
+  // Create a user for the task if the user does not exist
+  let user = await userRepository.findUserByEmail(sdk, email);
+
+  if (!user) {
+    console.log(chalk.white('⚙️  Creating a user for the task'));
+
+    user = await sdk.users.createAccount({
+      firstName: `${taskName}`,
+      lastName: 'exh.tasks',
+      email,
+      password,
+      phoneNumber: '0000000000',
+      language: 'EN',
+    });
+
+    console.log(chalk.green('✅ Successfully created a user for task'));
+
+    await assignRoleToUser(sdk, user.id, role.id);
+    return await createOAuth1Tokens(sdk, email, password);
+  }
+
+  // Check if the there are existing user credentials in the Function's environment variables
+  console.log(chalk.white('⚙️  Checking for the existing users credentials'));
+
+  const currentFunction = await functionRepository.findByName(sdk, taskName);
+
+  const hasExistingCredentials = (
+    currentFunction?.environmentVariables?.API_HOST?.value &&
+    currentFunction?.environmentVariables?.API_OAUTH_TOKEN_SECRET?.value &&
+    currentFunction?.environmentVariables?.API_OAUTH_TOKEN?.value &&
+    currentFunction?.environmentVariables?.API_OAUTH_CONSUMER_KEY?.value &&
+    currentFunction?.environmentVariables?.API_OAUTH_CONSUMER_SECRET?.value
+  );
+
+  // If there are no existing credentials, remove the user and create a new one
+  if (!hasExistingCredentials) {
+    console.log(chalk.white('⚙️  No existing credentials found for the user, removing the user'));
+
+    await sdk.users.remove(user.id);
```

Comment:
I'm not really comfortable with deleting a user (comfortable with only very few automatic resource deletion for that matter). Would it make sense to simply throw if a user exist but no credentials are found?